example/bookstore/v1/bookstore.yaml

```diff
@@ -8,18 +8,11 @@ resources:
         type: STRING
         number: 1
         required: true
-      price:
-        type: FLOAT
+      details:
+        type: OBJECT
         number: 2
         required: true
-      published:
-        type: BOOLEAN
-        number: 3
-        required: true
-      edition:
-        type: INT32
-        number: 4
-        required: false
+        object_type: Details
```

Comment:
I think this is a good short-term approach, but long-term I think people will want ways to define an object inline similar to the nicety of JSONSchema, vs purely reference based here. It will get cumbersome especially for deep nesting.

How about cutting a ticket to revisit this before advertising for public use? there's value in making this iterative step of just having objects at all.

Reply:
I've been going back and forth on this. In OAI, it seems really easy to create a lot of duplication this way. 

I agree we shouldn't block on this though.